src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs

```diff
@@ -68,10 +68,12 @@ public override async Task RunAsync(CancellationToken cancellationToken)
                 references.Add(name, bound);
             }
             Dictionary<string, SecretProperties> existingSecrets = (await storage.ListSecretsAsync()).ToDictionary(p => p.Name);
-            foreach (var (name, secret) in manifest.Secrets)
+
+            List<(string name, SecretManifest.Secret secret, SecretType.Bound bound)> orderedSecretTypes = GetTopologicallyOrderedSecrets(manifest.Secrets);
```

Comment:
For this to work exactly right, we also need to make sure that if we regenerate values for one secret, every secret depending on it is regenerated.

Reply:
Good point, I updated it, now all dependent secrets are regenerated. I also added some unit tests.